common/trustdomain_config.go

```diff
@@ -1,35 +1,90 @@
 package common
 
+import (
+	"crypto/rand"
+	"encoding/json"
+	"golang.org/x/crypto/nacl/box"
+)
+
 type TrustDomainConfig struct {
-	name          string
-	address       string
-	isValid       bool
-	isDistributed bool
+	Name          string
+	Address       string
+	IsValid       bool
+	IsDistributed bool
+	PublicKey     [32]byte
+	privateKey    [32]byte
+}
+
+type PrivateTrustDomainConfig struct {
```

Comment:
A server, after generating a keypair, does need to be able to save the private key to disk, though that isn't the default behavior. This struct, accessed by `.Private()`, allows the private key to be exported, such that marshaling includes the private key. The test in this PR shows the different use cases.